### ISSUE TITLE
Error: Duplicate list element ID

### ISSUE THREAD

**Issue by lightsprint09** [automerge/automerge#145]
Hi, thanks for this AMAZING tool you built. I use automerge for about 2 months now and the only error I get it the following:

`Error: Duplicate list element ID CD75A6BC-3D5D-4651-98D8-492EFB25D3E4:7`

It occurs randomly during editing a document. I don't know how to reproduce it yet. I would like to know what the error means and how it affects by use of automerge. 

Once I know where to look, I can provide more detailed informations.

**Comment by dnzz** [comment]
We had the same problem when we are setting the actor id. Basically when you add/remove/add an item to an array it will fail when you use Automerge.load with actorId.

Steps to reproduce:
1- Load with actorId
2- Add item to an array to your object
3- ApplyChanges
4- Save
5- Load the saved document again with the same actorId
6- Delete the item you added
7- ApplyChanges
8- Save
9- Load the saved document again with the same actorId
10- Add the item again
11- Apply changes will fail.

So we have removed actorId and it started working again.

**Comment by ept** [comment]
Thank you for the description! I have been able to reproduce the issue with the following code:

```js
let s = Automerge.init('actorid')
s = Automerge.change(s, doc => doc.list = ['a'])
s = Automerge.change(s, doc => doc.list.deleteAt(0))
s = Automerge.load(Automerge.save(s), 'actorid')
s = Automerge.change(s, doc => doc.list.push('b'))
```

The thrown exception is as follows:

```
Error: Duplicate list element ID actorid:1
    at applyInsert (/Users/martin/dev/cl/automerge/backend/op_set.js:86:72)
    at applyOps (/Users/martin/dev/cl/automerge/backend/op_set.js:242:25)
    at applyChange (/Users/martin/dev/cl/automerge/backend/op_set.js:267:21)
    at applyQueuedOps (/Users/martin/dev/cl/automerge/backend/op_set.js:285:26)
    at Object.addChange (/Users/martin/dev/cl/automerge/backend/op_set.js:332:23)
    at apply (/Users/martin/dev/cl/automerge/backend/index.js:146:36)
    at Object.applyLocalChange (/Users/martin/dev/cl/automerge/backend/index.js:184:23)
    at makeChange (/Users/martin/dev/cl/automerge/frontend/index.js:95:56)
    at Object.change (/Users/martin/dev/cl/automerge/frontend/index.js:262:12)
    at Object.change (/Users/martin/dev/cl/automerge/src/automerge.js:26:37)
```

I am looking into fixing it now.

**Comment by lightsprint09** [comment]
Any news on this issue

**Comment by ept** [comment]
This fix is part of the freshly released Automerge version 0.10.0 on npm.
